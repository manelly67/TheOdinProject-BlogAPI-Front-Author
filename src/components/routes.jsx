import App from "../App";

const routes = [
  {
    index: true,
    path: "/",
    element: <App />,
  },
];

export default routes;
