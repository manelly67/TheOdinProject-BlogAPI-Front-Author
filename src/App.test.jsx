import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { createBrowserRouter, RouterProvider } from "react-router-dom";
import routes from "./components/routes.jsx";
import { act } from "react";

describe("something truthy and falsy", () => {
  it("true to be true", () => {
    expect(true).toBe(true);
  });

  it("false to be false", () => {
    expect(false).toBe(false);
  });
});

describe("App component", () => {
  it("renders correct nav bar", () => {
    const router = createBrowserRouter(routes);
    render(<RouterProvider router={router} />);
    const lenghNavBar = screen.getAllByRole("link").length;
    expect(lenghNavBar).toBe(2);
  });
  it("renders correct initial page", () => {
    const router = createBrowserRouter(routes);
    render(<RouterProvider router={router} />);
    const { container } = render(<RouterProvider router={router} />);
    expect(container).toMatchSnapshot();
  });
  it("should display fetched data", async () => {
    let blogdata;
    async function getData(arg) {
      try {
        const response = await fetch(arg, { mode: "cors" });
        const responseData = await response.json();
        return responseData;
      } catch (error) {
        alert("Something was wrong. try again later");
        console.log(error);
      }
    }
   
    await act(async () => {
      blogdata = await getData("https://top-backend-blogapi.onrender.com/");
      expect(blogdata.title).not.toMatch("TITLE");
      expect(blogdata.allPosts).not.toBeUndefined();
    });
  });
});
