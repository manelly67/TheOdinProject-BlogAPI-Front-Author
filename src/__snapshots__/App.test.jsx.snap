// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders correct initial page 1`] = `
<div>
  <nav>
    <div>
      <a
        data-discover="true"
        href="/"
      >
        SIGN UP
      </a>
    </div>
    <div>
      <a
        data-discover="true"
        href="/"
      >
        LOGIN
      </a>
    </div>
  </nav>
  <div
    class="blog-content"
  >
    <h2>
      website for authoring and editing posts
    </h2>
    <p>
       
       
    </p>
    <div>
      Loading...
    </div>
  </div>
</div>
`;
